docker-compose.yml

```
@@ -1,13 +1,82 @@
+# This docker-compose.yml sets up a deployment of Mastodon's web (Puma), Sidekiq, and streaming containers,
+# as well as the required Postgres and Redis components. Optionally, you can enable Elastic Search and Tor by
+# removing the comments (#) from those blocks.
+
+# You should check the Docker image tag used for the web, sidekiq, and streaming containers to make sure you are running
+# the appropriate version for your deployment. Using `nightly` will deploy the in-development version of Mastodon, which is
+# not considered a stable release. Visit https://github.com/mastodon/mastodon/releases for currently released versions and tags.
+
 version: '3'
 services:
-  db:
+  web:
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:nightly
+    env_file: .env.production
     restart: always
-    image: postgres:14-alpine
-    shm_size: 256mb
+    command: bundle exec puma -C config/puma.rb
+    healthcheck:
+      test: ['CMD-SHELL', "curl -s --noproxy localhost localhost:3000/health | grep -q 'OK' || exit 1"]
     networks:
+      - external_network
       - internal_network
+    ports:
+      - '127.0.0.1:3000:3000'
+    volumes:
+      - ./public/system:/mastodon/public/system
+    depends_on:
+      - db
+      - redis
+      # - es
+
+  sidekiq:
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:nightly
+    env_file: .env.production
+    restart: always
+    command: bundle exec sidekiq
+    healthcheck:
+      test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+    networks:
+      - external_network
+      - internal_network
+    volumes:
+      - ./public/system:/mastodon/public/system
+    depends_on:
+      - db
+      - redis
+      # - es
+
+  streaming:
+    # build:
+    #   context: ./streaming
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon-streaming:nightly
+    env_file: .env.production
```

```suggestion
    env_file: .env.production
    environment:
      - PORT=4000
      - BIND=127.0.0.1
```

The `environment` values will take precedence over the `env_file`

---

Oh I hadn't thought about just doing that, I was thinking the `docker-compose.yml` would need to read from the env file for the healthcheck as some kind of variable. You're just suggesting we override that so it's not a potential conflict.

---

Yup, because PORT applies to both web and streaming, so what does `port` in the `.env.production` file actually mean?

You could do `WEB_PORT` and `STREAMING_PORT` (same for `BIND`, I wanna remove unix socket support, since I don't think anyone uses it)

---

Ticket for removing unix socket support is here: https://github.com/mastodon/mastodon/issues/29066